webrender/res/brush_image.glsl

```diff
@@ -79,6 +81,26 @@ void brush_vs(
     vec2 uv0 = res.uv_rect.p0;
     vec2 uv1 = res.uv_rect.p1;
 
+    RectWithSize local_rect = prim_rect;
+    vec2 stretch_size = image_data.stretch_size;
+
+    // If this segment should interpolate relative to the
+    // segment, modify the parameters for that.
+    if ((brush_flags & BRUSH_FLAG_SEGMENT_RELATIVE) != 0) {
+        local_rect = segment_rect;
+        stretch_size = local_rect.size;
+
+        if ((brush_flags & BRUSH_FLAG_SEGMENT_REPEAT_X) != 0) {
+            stretch_size.x = texel_rect.z - texel_rect.x;
```

Comment:
I thought that `stretch_size` was in layout space while `texel_rect` in device space, how does this work?

Reply:
Since the source image is provided by the user, it doesn't depend on device pixel ratio - it effectively maps one texel to one logical pixel, which (I think) is what we want.